plugins/flatpak/gs-flatpak.c

```diff
@@ -3178,6 +3184,66 @@ gs_flatpak_create_runtime_repo (GsFlatpak *self,
 	return g_steal_pointer (&app);
 }
 
+GsApp *
+gs_flatpak_create_app_from_repo_dir (GsFlatpak *self,
+				     GFile *file,
+				     GCancellable *cancellable,
+				     GError **error)
+{
+	g_autoptr(GPtrArray) remotes = NULL;
+	g_autofree gchar *uri = NULL;
+	gboolean dir_has_repo = FALSE;
+	gboolean reload_overview = FALSE;
+	GsApp *app = NULL;
+	FlatpakRemoteType usb_type[] = { FLATPAK_REMOTE_TYPE_USB };
+
+	remotes = flatpak_installation_list_remotes_by_type (self->installation,
+							     usb_type,
+							     G_N_ELEMENTS (usb_type),
+							     cancellable, error);
+	if (remotes == NULL)
+		return NULL;
+
+	uri = g_file_get_uri (file);
+
+	for (guint i = 0; i < remotes->len; ++i) {
+		FlatpakRemote *remote = g_ptr_array_index (remotes, i);
+		g_autofree gchar *url = flatpak_remote_get_url (remote);
+		g_autoptr(GError) local_error = NULL;
+
+		/* skip if the remote does not have the dir's path as a prefix */
+		if (!g_str_has_prefix (url, uri))
```

Comment:
The use of `uri` and `url` as separate variables is a bit confusing. How about renaming `uri` to `file_uri`, and `url` to `remote_uri`?

Also, be wary of string comparisons on URIs, since you can get false positives if the prefix doesn’t end in a slash. Use `g_file_has_parent()` instead?

Reply:
Good catch! Thanks.